.github/actions/setup-go/action.yml

```diff
@@ -0,0 +1,26 @@
+name: "Setup Go"
+description: "Setup Go with caching"
+inputs:
+  cache-prefix:
+    description: "Prefix for the cache key"
+    required: true
+outputs:
+  go-version:
+    description: "The installed Go version"
+    value: ${{ steps.setup-go.outputs.go-version }}
+runs:
+  using: "composite"
+  steps:
+    - id: setup-go
+      uses: actions/setup-go@v4
+      with:
+        go-version: "1.21"
```

Comment:
You can parametrize the go version and keep this as a default.

```
  go-version:
    description: "Go version to install"
    required: false
    default: "1.21"
```

Reply:
yes but I don't want to have to change it in multiple places, which is why I "hard-coded" it here 😄